src/main/java/com/google/sps/data/BusinessesService.java

```diff
@@ -77,12 +77,19 @@ public BusinessesService(List<Listing> allBusinesses) {
             .apiKey(KEY)
             .build();
     TextSearchRequest request = PlacesApi.textSearchQuery(context, product);
-    
+    int radius = 1500;
+
     try {
       PlacesSearchResponse response = request.location(latLng)
-              .radius(TEXT_SEARCH_RADIUS)
+              .radius(radius)
               .await();
-      
+
+      while (response.results.length < 20 && 
+             radius < MAX_ALLOWED_TEXT_SEARCH_RADIUS) {
+        radius *= 4;
```

Comment:
Please make 20 and 4 well-defined constants at the top. A comment may also be useful.

Reply:
fixed